src/plugins/intel_gpu/src/plugin/transformations/op/gemm.cpp

```diff
@@ -58,9 +108,45 @@ bool Gemm::visit_attributes(ov::AttributeVisitor &visitor) {
 
 std::vector<ov::PartialShape> shape_infer(const Gemm* op,
                                           std::vector<ov::PartialShape> input_shapes,
+                                          const std::vector<int32_t>& target_shape_a,
+                                          const std::vector<int32_t>& target_shape_b,
+                                          const std::vector<int64_t>& output_pattern_a,
+                                          const std::vector<int64_t>& output_pattern_b,
                                           const std::vector<int64_t>& order_a,
                                           const std::vector<int64_t>& order_b,
                                           const std::vector<int64_t>& order_c) {
+    auto shape_a = input_shapes[0];
+    auto shape_b = input_shapes[1];
+
+    // broadcasted shapes
+    auto broadcast_shape = [](const ov::PartialShape shape, const std::vector<int32_t>& target_shape) {
```

Comment:
That approach looks overcomplicated. Basically, this feature is supposed to cover grouped query attention in the first place. 

Current semantics of gemm supports broadcast for batch dimensions already which means that:
`out_shape=[B0, B1, ..., Bn, M, N] => (in_shape[i] == 1 or in_shape[i] == out_shape[i]) for i in [0, n-1].` 

So all we need for GQA support is to change first condition like this:
`out_shape[i] % in_shape[i] == 0 or in_shape[i] == out_shape[i] `

Shape inference update is trivial in this case too: 
`out_shape[i] = max(in0_shape[i], in1_shape[i]) for i in [0, n-1]`

Do I miss something? Let me know if there's a reason for such implementation

Reply:
As you suggested, can I proceed with shape inference update for GQA support by including it in the scope of the next task?